logrotate.c

```diff
@@ -510,8 +510,13 @@ static int createOutputFile(char *fileName, int flags, struct stat *sb,
     if (fchmod(fd, (S_IRUSR | S_IWUSR) & sb->st_mode)) {
 	message(MESS_ERROR, "error setting mode of %s: %s\n",
 		fileName, strerror(errno));
-	close(fd);
-	return -1;
+
+		if (geteuid() == 0) {
+			/* If run as non-root, we may not be able to duplicate
+			the ownership of the file. Warn but don't fail. */
```

Comment:
I agree that wording of the comment is not optimal, but it is not the main problem here...

I am afraid that you are changing a wrong block of the code.  It is ```fchown()``` what is expected to fail for a non-root user whereas you are patching the error handling of ```fchmod()```.